packages/orchestration/src/cosmos-api.ts

```diff
@@ -0,0 +1,246 @@
+import type { AnyJson } from '@agoric/cosmic-proto';
+import type {
+  Delegation,
+  Redelegation,
+  UnbondingDelegation,
+} from '@agoric/cosmic-proto/cosmos/staking/v1beta1/staking.js';
+import type {
+  MsgBeginRedelegateResponse,
+  MsgUndelegateResponse,
+} from '@agoric/cosmic-proto/cosmos/staking/v1beta1/tx.js';
+import type { TxBody } from '@agoric/cosmic-proto/cosmos/tx/v1beta1/tx.js';
+import type { Port } from '@agoric/network';
+import type {
+  LocalIbcAddress,
+  RemoteIbcAddress,
+} from '@agoric/vats/tools/ibc-utils.js';
+import type { ChainAddress } from './orchestration-api.js';
+import type { AmountArg, Chain, KnownChains } from './types.js';
+
+/** A helper type for type extensions. */
+export type TypeUrl = string;
+
+// TODO move into cosmic-proto
+export type Proto3JSONMsg = {
+  '@type': TypeUrl;
+  value: Record<string, unknown>;
+};
+
+/** An address for a validator on some blockchain, e.g., cosmos, eth, etc. */
+export type CosmosValidatorAddress = ChainAddress & {
+  // TODO document why this is the format
+  address: `${string}valoper${string}`;
+  addressEncoding: 'bech32';
+};
+
+/**
+ * A denom that designates a token type on some blockchain.
+ *
+ * Multiple denoms may designate the same underlying base denom (e.g., `uist`,
+ * `uatom`) on different Chains or on the same Chain via different paths. On
+ * Cosmos chains, all but the base denom are IBC style denoms, but that may vary
+ * across other chains. All the denoms that designate the same underlying base
+ * denom form an equivalence class, along with the unique Brand on the local
+ * Chain. Some operations accept any member of the equivalence class to
+ * effectively designate the corresponding token type on the target chain.
+ */
+export type Denom = string; // ibc/... or uist
+
+/** Description for an amount of some fungible currency */
+export type CoinAmount = {
```

Comment:
Also, "Coin" may imply too much specifics, which is why we used "ChainAmount".  Is this being renamed in this PR

Reply:
yes, by the logic that it was Cosmos-only, and that's what Cosmos types call it.

I'm putting it back to `ChainAmount`, but I'll note the mismatch with Agoric's "amount" that aren't all fungible.

it also seems to conflate amount naming with chain naming, whereas on Agoric there are many amounts for the one chain.

why not `DenomAmount`?

Reply:
actually, `NativeAmount` would be easily confused with _Agoric-native_ (ERTP) Amount.
```
/** Amounts can be provided as pure data using denoms or as native Amounts */
export type AmountArg = ChainAmount | Amount;
```
maybe `RemoteAmount` to convey the difference. Though that still sounds like a _kind_ of (ERTP) `Amount`.

I see the goal of naming these as like ERTP Amounts, but I think having "Amount" in the name of this not ERTP Amount is more confusing than helpful.

Reply:
Also having it be an Amount made `asAmount` confusing because it's ostensibly already an amount. That led me to omit Amount from the name. I'm pushing with `TokenCount` which conveys both that it's fungible and not an ERTP Amount.

```js
asAmount: (amount: TokenCount) => NatAmount;
```

Reply:
We agreed on DenomAmount